packages/zoltan2/test/color/coloring1.cpp

```diff
@@ -231,19 +355,19 @@ int main(int narg, char** arg)
   try
   {
   Zoltan2::ColoringProblem<SparseMatrixAdapter> problem(&adapter, &params);
-  std::cout << "Going to color" << std::endl;
+  if(comm->getRank()==0) std::cout << "Going to color" << std::endl;
   problem.solve();
 
   ////// Basic metric checking of the coloring solution
   size_t checkLength;
-  int *checkColoring;
+  int *checkColoring = NULL;
```

Comment:
More raw pointers.  Raw pointers are bad.

Reply:
This variable is used to store the returned value of the Zoltan2::ColoringSolution class's getColors() function.

Reply:
@ibogle wrote:
> This variable is used to store the returned value of the `Zoltan2::ColoringSolution` class's `getColors()` function.

Ah, so be it then.  If `getColors()` returns a new allocation that the caller is responsible for deallocating, then it should actually return `std::unique_ptr<int*>`, not `int*`.